flocker/node/agents/blockdevice.py

```diff
@@ -1004,6 +1010,8 @@ def create_volume(dataset_id, size):
             volume.
         :param int size: The size of the new volume in bytes.
         :returns: A ``BlockDeviceVolume``.
+        :raises: ``VolumeExists`` if a volume for the supplied ``dataset_id``
+            already exists in the cluster.
```

Comment:
Instead of making this part of the interface and making every implementation check, why not do `check_for_existing_dataset` at the top-ish of `CreateBlockDeviceDataset.run`?